themes/default/content/challenge/startup-in-a-box.md

```diff
@@ -508,9 +513,419 @@ expect(await page.title()).toBe("Pulumi Challenge");
+      GCP <a href="https://cloud.google.com/gcp" target="_blank" rel="noopener noreferrer">Account</a> & <a href="https://cloud.google.com/resource-manager/docs/creating-managing-projects/" target="_blank" rel="noopener noreferrer">Project</a>
+    </li>
+    <li>
+      <a href="https://www.checklyhq.com/" target="_blank" rel="noopener noreferrer">Checkly</a> account
+    </li>
+  </ul>
+
+  <p><b>Note:</b> New to Google Cloud? You can get $300 in free credits to run, test, and deploy workloads. <a href="https://cloud.google.com/gcp" target="_blank" rel="noopener noreferrer">Open a free GCP account</a> and use it for this Pulumi Challenge</p>
+</div>
+
+### Challenge
+
+Follow along with the steps outlined on this page to complete the challenge.
+
+#### Step 1. Your First Pulumi Program
+
+You will learn how to create a new Pulumi program using our Pulumi templates, specifically for GCP with TypeScript. Create a new directory called `pulumi-challenge` and run the following inside of it, following the prompts if requested:
+
+```shell
+  # set up environment
+  pulumi new gcp-typescript
+
+  # Set up your project id and possibly the region
+  pulumi config set gcp:project <GCP_PROJECT_ID>
+
+  # or whatever your porjectid is
+  pulumi config set gcp:region europe-west1
+```
+
+#### Step 2. Creating Your First Resource
+
+Now that we have a base GCP project configured, we need to create our first resource. In this instance, we’ll create a new [GCS bucket](https://cloud.google.com/storage/docs/buckets) which will allow us to store our static website. The command ```pulumi new gcp-typescript``` produced a a file ```pulumi-challenge/index.ts```. Clear the contents of the ```index.ts``` file and add the following script to create your first resources.
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as gcp from "@pulumi/gcp";
+
+// Create a GCP resource (Storage Bucket)
+const bucket = new gcp.storage.Bucket("mybucket", {
+    location: "EU",
+     website: {
+        mainPageSuffix: "index.html",
+    },
+},{});
+
+// Create an IAM binding to allow public read access to the bucket.
+const bucketIamBinding = new gcp.storage.BucketIAMBinding("bucket-iam-binding", {
+    bucket: bucket.name,
+    role: "roles/storage.objectViewer",
+    members: ["allUsers"],
+});
+```
+
+#### Step 3. Working with Local Files
+
+Pulumi lets you use your favourite programming language to define your infrastructure. Today, we’re using TypeScript, which means we have access to the Node API. This includes discovering directories and files.
+
+We can use a synced folder to manage the files of the website.
+
+We need to add the synced-folder package from npm, to easiyl upload files to the Cloud Storage Bucket.
+
+```cli
+npm install @pulumi/synced-folder
+```
+
+Now with the synced-folder package installed we need to update our pulumi code. Add the following to ```index.ts```
+
+```typescript
+import * as synced_folder from "@pulumi/synced-folder";
```

Comment:
is there a reason y'all went the synced-folder approach vs sticking with mime?

Reply:
just to make it more simply - BUT I agree to stick to standard Pulumi provider capabilities rather then using abstracted packages for this intro example. will adopt to new gcp.storage.BucketObject